src/libstd/io/mod.rs

```diff
@@ -287,15 +532,35 @@ pub trait Read {
 
 /// A trait for objects which are byte-oriented sinks.
 ///
-/// The `write` method will attempt to write some data into the object,
-/// returning how many bytes were successfully written.
+/// Implementors of the `Write` trait are sometimes called 'writers'.
+///
+/// Writers are defined by two required methods, `write()` and `flush()`:
 ///
-/// The `flush` method is useful for adaptors and explicit buffers themselves
-/// for ensuring that all buffered data has been pushed out to the "true sink".
+/// * The `write()` method will attempt to write some data into the object,
+///   returning how many bytes were successfully written.
 ///
-/// Writers are intended to be composable with one another. Many objects
-/// throughout the I/O and related libraries take and provide types which
-/// implement the `Write` trait.
+/// * The `flush()` method is useful for adaptors and explicit buffers
+///   themselves for ensuring that all buffered data has been pushed out to the
+///   'true sink'.
+///
+/// Writers are intended to be composable with one another. Many implementors
+/// throughout `std::io` take and provide types which implement the `Write`
+/// trait.
+///
+/// # Examples
+///
+/// ```
+/// use std::io::Write;
+/// use std::fs::OpenOptions;
+///
+/// # fn foo() -> std::io::Result<()> {
+/// let mut buffer = try!(OpenOptions::new().write(true).open("foo.txt"));
```

Comment:
You might be interested in the more ergonomic `File::create("foo.txt")` instead of going through OpenOptions :)